activerecord/test/cases/relations_test.rb

```diff
@@ -602,6 +602,11 @@ def test_preload_applies_to_all_chained_preloaded_scopes
     end
   end
 
+  def test_extracted_association
+    authors = Post.all.extract_associated(:author)
+    assert_equal Post.all.collect(&:author), authors
```

Comment:
Perhaps we should wrap the extract in an assert_queries 1, such that we test the preload query optimization.

Reply:
Good idea!